File: python/tree/BinaryTreeLr.py
from Queue import MyQueue
from BinaryTree import TreeNode

class BinaryTree:
    def __init__(self, root):
        self.root = root

    @staticmethod
    def levelOrderTraversal(root):
        queue = MyQueue(8)
        queue.enQueue(root)
        while queue.front != queue.rear:
            node = queue.deQueue()
            print(node.data)
            if node.leftChild is not None:
                queue.enQueue(node.leftChild)
            if node.rightChild is not None:  
                queue.enQueue(node.rightChild)


if __name__ == "__main__":
    # 创建一个包含多个节点的二叉树
    root = TreeNode(1) 
    root.leftChild = TreeNode(2)
    root.rightChild = TreeNode(3)
    root.leftChild.leftChild = TreeNode(4)
    root.leftChild.rightChild = TreeNode(5)
    root.rightChild.rightChild = TreeNode(6)
    #创建二叉树对象 
    tree = BinaryTree(root)

    print("二叉树层序遍历：")
    BinaryTree.levelOrderTraversal(tree.root)

File: C++/tree/BinaryTreeLr.cpp
#include<iostream>
#include"BinaryTree.h"
using namespace std;

class BinaryTreeLr{
    public:
    int height(TreeNode* root){
        if(root ==nullptr){
            return 0;
        }else{
            int left_height = height(root->leftChild);
            int right_height =height(root->rightChild);
            return max(left_height,right_height) +1;
        }
    }
    void printLevelOrder(TreeNode* root,int level){
        if(root == nullptr){
            return ;
        }
        if(level ==1){
            cout <<root->data<<" ";
        }else if(level >1){
            printLevelOrder(root->leftChild,level -1);
            printLevelOrder(root->rightChild,level -1);
        }
    }
    void levelOrderRecursive(TreeNode* root){
        int h =height(root);
        for(int i = 1;i<=h;i++){
            printLevelOrder(root,i);
        }
    }
};

int main(){
    // 创建一个包含多节点的二叉树
    TreeNode* root = new TreeNode(1);
    root->leftChild = new TreeNode(2);
    root->rightChild = new TreeNode(3);
    root->leftChild->leftChild = new TreeNode(4);
    root->leftChild->rightChild = new TreeNode(5);
    root->rightChild->rightChild = new TreeNode(6);

    BinaryTreeLr binaryTreeLr; // 实例化 BinaryTreeLr 类
    cout << "层序递归遍历：" << endl;
    binaryTreeLr.levelOrderRecursive(root); 

    // 释放内存空间
    delete root->leftChild->leftChild;
    delete root->leftChild->rightChild;
    delete root->rightChild->rightChild;
    delete root->leftChild;
    delete root->rightChild;
    delete root;

    return 0;
}
